### DISCUSSION TITLE
Custom Trajectories

### DISCUSSION THREAD

**Discussion by jblich870** [GreenleafLab/ArchR#1689] [Feature Requests]
Hi,

How and where are trajectory coordinates stored within ArchR projects? 

And, would it be possible to import trajectory co-ordinates generated elsewhere into an ArchR project for downstream analyses (for example use of plotting, getTrajectory and correlateTrajectories)? 

Alternatively, would it be possible to include more of the native functionality of Slingshot in the ArchR Slingshot wrapper? For example, specifying end groups, and use of the "extend" parameter.

I have generated Slingshot trajectories using the Slingshot function but I can't reproduce them using the ArchR Slingshot function, which instead gives sub-optimal trajectories. Ideally I would use the Slingshot-generated trajectories and import them into ArchR project for downstream analyses.

Many thanks,
J

**Comment by annrosebright** [comment]
Hi, 

I have a similar question as well. 

a) Can we import and use the external pseudotime to plot motif activity and expression of Multiome data? The analysis I would like to perform is to check if TF expression and motif activity correlates across the clusters or if are there instances where motif activity precedes the expression of TF. 

Best,
Ann

**Comment by rcorces** [comment]
I will get to this soon but am behind as I've been traveling. It might take me another week to get caught back up.

**Reply by jblich870** [reply]
Apologies for delayed reply, I've also been away.

Many thanks for your reply. Would be great to hear your thoughts on this.

**Reply by rcorces** [reply]
Still behind but this is on my radar

**Comment by rcorces** [comment]
@jblich870 
> How and where are trajectory coordinates stored within ArchR projects?

Trajectories are stored as columns in `cellColData`. In ArchR, a trajectory is just a collection of pseudotime values, ranging from 0 to 100 (decimals allowed) which provide the explicit ordering of cells across the trajectory.

> And, would it be possible to import trajectory co-ordinates generated elsewhere into an ArchR project for downstream analyses (for example use of plotting, getTrajectory and correlateTrajectories)?

I dont see why this wouldnt be possible, you just need to create a column in `cellColData` that maps each cell to a pseudotime value between 0 and 100.

> Alternatively, would it be possible to include more of the native functionality of Slingshot in the ArchR Slingshot wrapper? For example, specifying end groups, and use of the "extend" parameter.

I dont see why not. Slingshot and Monocle3 trajectories were added on as a bonus and havent really been fleshed out. If you can provide me with explicit examples of what you're trying to accomplish (bonus points if this is done using the tutorial data so that it is easy for me to reproduce), then I can look into it.

@annrosebright 
> a) Can we import and use the external pseudotime to plot motif activity and expression of Multiome data?

As in my answer above, I dont see why importing external pseudotime wouldnt work as long as you can match the format expected by ArchR. Its really just another column in `cellColData` with pseudtime values.

Hope that helps

**Reply by jblich870** [reply]
Hi,

Thanks for the reply.

I suppose what I meant by 'trajectory coordinates' was coordinates of the actual aligned trajectory, i.e. the 'continuous trajectory' described in the manual: "ArchR fits a continuous trajectory to each N-dimensional coordinate based on the pseudo-time value using the smooth.spline function". I assume this is what is represented by the arrow produced by the `plotTrajectory` function?

If these are stored somewhere, I thought I could replace them with custom coordinates generated elsewhere, and then generate pseudotime values for each cell aligned to this new trajectory.

I posted a separate but potentially related issue: https://github.com/GreenleafLab/ArchR/discussions/1729

I'll have a think about specific examples for including more native Slingshot functions and get back to you, thanks.

**Reply by rcorces** [reply]
see https://github.com/GreenleafLab/ArchR/discussions/1729#discussioncomment-4179548

**Reply by jblich870** [reply]
Regarding native Slingshot functions, what would be particularly helpful would be:

1. Ability to generate branched trajectories, that could be supervised for example by specifying a starting cluster and multiple end clusters

2. Output containing a weight that assigns each cell to one or more lineages (for example as extracted by `slingCurveWeights` in the Slingshot package https://bioconductor.org/packages/devel/bioc/vignettes/slingshot/inst/doc/vignette.html#constructing-smooth-curves-and-ordering-cells)

These together would enable use of TradeSeq which allows statistical comparisons between trajectories. (This has been raised before e.g. https://github.com/GreenleafLab/ArchR/discussions/1495 https://github.com/GreenleafLab/ArchR/issues/190).

Would be very grateful if you would consider adding this. Let me know if you need any further info.

Best wishes,
J
